src/rgw/rgw_rados.cc

```diff
@@ -3937,6 +4009,21 @@ int RGWRados::fetch_remote_obj(RGWObjectCtx& obj_ctx,
     set_mtime_weight.init(set_mtime, svc.zone->get_zone_short_id(), pg_ver);
   }
 
+  if (cct->_conf->rgw_sync_obj_etag_verify) {
```

Comment:
as above, this shouldn't depend on rgw_sync_obj_etag_verify either
```c++
std::string calculated_etag = cb.get_calculated_etag();
if (!calculated_etag.empty()) {
```

Reply:
I think I will fallback on `obj_type` member in `class RGWRadosPutObj` to find out if we need to perform etag verification.